plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -140,14 +144,22 @@ public static TableStatistics makeTableStatistics(
                 .map(column -> Maps.immutableEntry(column.fieldId(), column.type()))
                 .collect(toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
 
+        Domain pathDomain = getPathDomain(effectivePredicate);
+        effectivePredicate = effectivePredicate.filter((icebergColumnHandle, domain) -> !icebergColumnHandle.equals(pathColumnHandle()));
         TableScan tableScan = icebergTable.newScan()
                 .filter(toIcebergExpression(effectivePredicate))
                 .useSnapshot(snapshotId)
                 .includeColumnStats();
 
         IcebergStatistics.Builder icebergStatisticsBuilder = new IcebergStatistics.Builder(columns, typeManager);
         try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
-            fileScanTasks.forEach(fileScanTask -> icebergStatisticsBuilder.acceptDataFile(fileScanTask.file(), fileScanTask.spec()));
+            fileScanTasks.forEach(fileScanTask -> {
```

Comment:
Could you add a test?

Reply:
Could you suggest where should the test be added? And do you have suggestion about how the test would looks like?